src/pyodide/internal/python.js

```diff
@@ -419,20 +419,27 @@ function setUploadFunction(toUpload) {
   };
 }
 
+// "\x00snp"
+const SNAPSHOT_MAGIC = 0x6d736100;
+const SNAPSHOT_VERSION = 1;
+
 /**
  * Encode heap and dsoJSON into the memory snapshot artifact that we'll upload
  */
 function encodeSnapshot(heap, dsoJSON) {
   const dsoString = JSON.stringify(dsoJSON);
-  let sx = 8 + 2 * dsoString.length;
+  let snapshotOffset = 8 + 2 * dsoString.length;
   // align to 8 bytes
-  sx = Math.ceil(sx / 8) * 8;
-  const toUpload = new Uint8Array(sx + heap.length);
+  snapshotOffset = Math.ceil(snapshotOffset / 8) * 8;
+  const toUpload = new Uint8Array(snapshotOffset + heap.length);
   const encoder = new TextEncoder();
   const { written } = encoder.encodeInto(dsoString, toUpload.subarray(8));
+  const jsonLength = written;
```

Comment:
```suggestion
  const { jsonLength } = encoder.encodeInto(dsoString, toUpload.subarray(8));
```

Reply:
Well this change won't work, but you can do `written: jsonLength`.